stdlib/http/src/main/java/org/ballerinalang/net/http/actions/httpclient/AbstractHTTPAction.java

```diff
@@ -358,10 +357,20 @@ private void send(DataContext dataContext, HttpCarbonMessage outboundRequestMsg,
             future.setHttpConnectorListener(httpClientConnectorLister);
         }
         try {
-            if (boundaryString != null) {
-                serializeMultiparts(dataContext.context, messageOutputStream, boundaryString);
-            } else {
-                serializeDataSource(dataContext.context, messageOutputStream);
+            BMap<String, BValue> requestStruct = ((BMap<String, BValue>) dataContext.context.getNullableRefArgument(1));
+
+            if (requestStruct != null) {
+                BMap<String, BValue> entityStruct = extractEntity(requestStruct);
+                if (entityStruct != null) {
+                    if (boundaryString != null) {
+                        serializeMultiparts(entityStruct, messageOutputStream, boundaryString);
+                    } else {
+                        serializeDataSource(entityStruct, messageOutputStream);
+                    }
+                } else {
+                    //This is not really a passthrough but is the case when the body is null/not built.
```

Comment:
This comment is a bit ambiguous. Can we rephrase it to give the exact meaning as to why it is set as passthrough? Something along the line that it reaches here when it is a genuine passthrough (the body has not been read) or the entity body is actually null. Since there's no way to distinguish between the two from ballerina side, it is always set as true but it does no harm and transport side will interpret this value only when there's an unread body in carbon message.